### DISCUSSION TITLE
Filtering specific columns within a Global Filter

### DISCUSSION THREAD

**Discussion by kaeland** [TanStack/table#2722] [General]
Hi there, 

I'm currently attempting to build an Airtable-esque filter component. I'm wondering if it is possible to use the `useGlobalFilter` hook to choose what columns to filter a table by programmatically. For example, Airtable implements table filtering using a component that lives outside of a given table's boundaries. Inside of this component, there is a dropdown menu to choose which column to apply a `filterValue` and a dropdown to choose the `filterType` for a given column. Based on react-table's documentation, it appears that `useGlobalFilter` is used to build filtering components that exist outside of the table. However, based on the documentation's code sandbox example, for filtering, it also appears that `useGlobalFilter` applies the filter value to all columns rather than to specific columns.  

My question would be if it's possible to use `useGlobalFilter`'s ability to create filtering UI outside of a table and have a way to select what columns to apply a `filterValue` & `filterType`, all from within the global filter? 

If this is possible, would anyone be willing to provide tips or advice on the implementation? Please let me know if this task would be more suited for `useFilters` or another part of react-table's API. 

I've provided a screenshot of Airtable's filter component as an example of what I'd like to build. Any feedback or advice would be much appreciated! ☺️

<img width="1434" alt="Screen Shot 2020-09-10 at 6 49 45 PM" src="https://user-images.githubusercontent.com/17867788/92944447-976bb700-f419-11ea-9c4f-822cdf6ee41f.png">

<img width="1434" alt="Screen Shot 2020-09-10 at 6 50 03 PM" src="https://user-images.githubusercontent.com/17867788/92944516-af433b00-f419-11ea-8357-58d8e9798ccb.png">

**Comment by 000000armor** [comment]
Sorry for offtopic but this app is fire bro!

**Reply by kaeland** [reply]
Airtable is pretty fire!

**Comment by HermanNygaard** [comment]
I think `setFilter` would accomplish this, just call it on multiple column ID's: [Here is a CodeSandbox with two examples](https://codesandbox.io/s/react-table-filter-outside-table-bor4f?file=/src/App.js)

**Reply by kaeland** [reply]
Hey, thank you!

**Reply by raphaelfavier** [reply]
Did `setFilter` changed since last year? I can't figure out how to call it on multiple column ID's

A solution I found is this [one](https://stackoverflow.com/questions/69177371/react-table-disable-global-filter-for-one-column-but-add-another-filter-for-d)
